File: src/main/java/fi/aalto/cs/apluscourses/intellij/notifications/CourseConfigurationError.java
package fi.aalto.cs.apluscourses.intellij.notifications;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
import org.jetbrains.annotations.NotNull;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getAndReplaceText;
import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

public class CourseConfigurationError extends Notification {

  private final Exception exception;

  /**
   * Constructs a notification that notifies the user of an error that occurred while attempting to
   * parse a course configuration file.
   *
   * @param exception An exception that caused this notification.
   */
  public CourseConfigurationError(@NotNull Exception exception) {
    super(
        PluginSettings.A_PLUS,
        getText("notification.CourseConfigurationError.title"),
        getAndReplaceText("notification.CourseConfigurationError.content",
            exception.getMessage()),
        NotificationType.ERROR);
    this.exception = exception;
  }

  public Exception getException() {
    return exception;
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/notifications/CourseConfigurationError.kt
package fi.aalto.cs.apluscourses.notifications

import com.intellij.notification.Notification
import com.intellij.notification.NotificationType
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.services.PluginSettings

/**
 * A notification that notifies the user of an error that occurred while attempting to
 * parse a course configuration file.
 *
 * @param exception An exception that caused this notification.
 */
class CourseConfigurationError(val exception: Exception) : Notification(
    PluginSettings.A_PLUS,
    MyBundle.message("notification.CourseConfigurationError.title"),
    MyBundle.message(
        "notification.CourseConfigurationError.content",
        exception.message!!
    ),
    NotificationType.ERROR
)
